### PR TITLE
Parametrize: fix Amber improper reading from a prepi file

### LINKED ISSUE #391: FFEvaluate energy of torsion term does not match with OpenMM
Get aniline (Ph-NH2) from https://raw.githubusercontent.com/Acellera/parameterise-testset/master/reference/m051/input.mol2

Calculate energy of aniline with HTMD:
```python
from htmd.parameterization.ffmolecule import FFMolecule
from htmd.parameterization.fftype import FFTypeMethod
from htmd.parameterization.ffevaluate import FFEvaluate

mol = FFMolecule('input.mol2', method=FFTypeMethod.GAFF2)
res = FFEvaluate(mol).evaluate(mol.coords[:,:,0])
print(res)

typemap = mol._prm.writeFrcmod(mol._rtf, 'mol.frcmod')
mol.write('mol.mol2', typemap=typemap)
``` 
```
{'elec': -2.051702647788332, 'vdw': 3.9122669768070497, 'bond': 0.77021301246485774, 'angle': 0.4833635420588632, 'dihedral': 1.612904528740123, 'improper': 0.0, 'total': 4.7270454122825614}
```

Generate *prmtop* and *inpcrd* files
```bash
$ cat tleap.inp
loadAmberParams mol.frcmod
MOL = loadMol2 mol.mol2
saveAmberParm MOL mol.prmtop mol.inpcrd
quit
$ tleap -f tleap.inp
```

Calculate energy of aniline with *ParmEd/OpenMM*:
```python
import parmed
from simtk import openmm 
from simtk import unit

prmtop = parmed.amber.LoadParm('mol.prmtop', xyz='mol.inpcrd')
system = prmtop.createSystem()
integrator = openmm.LangevinIntegrator(300 * unit.kelvin, 1 / unit.picoseconds, 2 * unit.femtoseconds)
platform = openmm.Platform.getPlatformByName('CPU')
context = openmm.Context(system, integrator, platform)
context.setPositions(prmtop.positions)
res = parmed.openmm.energy_decomposition(prmtop, context)
print(res)
```
```
{'bond': 0.7702162476656408, 'angle': 0.48337638371315556, 'dihedral': 2.8833159558052577, 'nonbonded': 1.8605609115871495, 'total': 5.997469498771204}
```

The energy of dihedral term does not match:
```
HTMD: 1.612904528740123
OpenMM: 2.8833159558052577
```

### PR DESCRIPTION
**Pull request by raimis**
Fixes partially #391.

### CODE REVIEW

**Comment by j3mdamas** [comment]
Updating branch, reviewing soon.

**Comment by stefdoerr** [comment]
I have little clue on parametrization. I'll leave this to Joao

File: htmd/parameterization/ff.py [hunk 1]
```diff
@@ -688,6 +709,17 @@ def __init__(self, mol, prepi, frcmod):
...
+            text = file.read()
+        impropers = re.search('^IMPROPER\n(.+)\n\n', text, re.MULTILINE | re.DOTALL)  # extract improper section
+        if impropers:
+            impropers = impropers.group(1).split('\n')  # array of improper lines
+            impropers = [improper.split() for improper in impropers]  # impropers by names
+            for improper in impropers:
+                improper_indices = [self.index_by_name[name.upper()] for name in improper]  # conv atom name to indices
+                self.impropers.append(improper_indices)
+
         f = open(frcmod, "r")
```
Comment (stefdoerr) [inline htmd/parameterization/ff.py]: If it's a file reader it should be moved to htmd/molecule/readers.py

Comment (raimis) [inline htmd/parameterization/ff.py]: Not only this. All the file is dealing with the reading and writing of RTF and PRM files!

Comment (j3mdamas) [inline htmd/parameterization/ff.py]: I agree Stefan, but this is a sort of reader of `FFMolecule`, which sub-classes `Molecule`. Unless we want to have `Molecule` reading force-field files, but that would move functionality of  `FFMolecule` to `Molecule`, which may be what we want, I don't know.
For now, this does not prevent the fix. I'm merging it.

**Comment by stefdoerr** [comment]
Just saying that readers should not just be thrown in random classes
methods. Should be organized in separate file as functions.

**Comment by j3mdamas** [comment]
You are right and I agree with you :)
The initial reader was not created by Raimondas, he's just fixing the current reader, not refactoring it

**Pull request merged**